packages/flutter_tools/lib/src/web/web_device.dart

```diff
@@ -177,27 +157,139 @@ class ChromeDevice extends Device {
   }
 }
 
+/// The Google Chrome browser based on Chromium.
+class GoogleChromeDevice extends ChromiumDevice {
+  GoogleChromeDevice({
+    @required Platform platform,
+    @required ProcessManager processManager,
+    @required ChromiumLauncher chromiumLauncher,
+    @required Logger logger,
+    @required FileSystem fileSystem,
+  }) : _platform = platform,
+       _processManager = processManager,
+       super(
+          name: 'chrome',
+          chromeLauncher: chromiumLauncher,
+          logger: logger,
+          fileSystem: fileSystem,
+       );
+
+  final Platform _platform;
+  final ProcessManager _processManager;
+
+  @override
+  String get name => 'Chrome';
+
+  @override
+  Future<String> get sdkNameAndVersion async => _sdkNameAndVersion ??= await _computeSdkNameAndVersion();
+
+  String _sdkNameAndVersion;
+  Future<String> _computeSdkNameAndVersion() async {
+    if (!isSupported()) {
+      return 'unknown';
+    }
+    // See https://bugs.chromium.org/p/chromium/issues/detail?id=158372
+    String version = 'unknown';
+    if (_platform.isWindows) {
+      final ProcessResult result = await _processManager.run(<String>[
+        r'reg', 'query', r'HKEY_CURRENT_USER\Software\Google\Chrome\BLBeacon', '/v', 'version',
+      ]);
+      if (result.exitCode == 0) {
+        final List<String> parts = (result.stdout as String).split(RegExp(r'\s+'));
+        if (parts.length > 2) {
+          version = 'Google Chrome ' + parts[parts.length - 2];
+        }
+      }
+    } else {
+      final String chrome = chromeLauncher.findExecutable();
+      final ProcessResult result = await _processManager.run(<String>[
+        chrome,
+        '--version',
+      ]);
+      if (result.exitCode == 0) {
+        version = result.stdout as String;
+      }
+    }
+    return version.trim();
+  }
+
+}
+
+/// The Microsoft Edge browser based on Chromium.
+class MicrosoftEdgeDevice extends ChromiumDevice {
```

Comment:
The edge device isn't hooked up yet

Reply:
Should we note somewhere in the code that this isn't hooked up yet?

Reply:
Done, here and in the validator